### PR TITLE
Internal: Conversion Center - add contact button document

### PR DESCRIPTION
**Pull request by nicoladj77**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Implement Contact Button Document

## Description
An explanation of what is done in this PR

* Implement Contact Button Document

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: assets/dev/js/editor/components/template-library/views/parts/templates.js [hunk 1]
```diff
@@ -153,7 +153,7 @@ TemplateLibraryCollectionView = Marionette.CompositeView.extend( {
 	},
 
 	setFiltersUI() {
-		if ( ! this.select2Instance ) {
+		if ( ! this.select2Instance && this.$( this.ui.selectFilter ).length ) {
```
Comment (nicoladj77) [inline assets/dev/js/editor/components/template-library/views/parts/templates.js:156]: @louiswol94 this avoids error when categories are not returned.

File: app/modules/import-export/utils.php [hunk 2]
```diff
@@ -81,7 +81,8 @@ public static function get_registered_cpt_names() {
 		unset(
 			$post_types[ Landing_Pages_Module::CPT ],
 			$post_types[ Source_Local::CPT ],
-			$post_types[ Conversion_Center_Module::CPT_LINKS_PAGES ]
+			$post_types[ Conversion_Center_Module::CPT_LINKS_PAGES ],
```
Comment (louiswol94) [inline app/modules/import-export/utils.php:84]: @nicoladj77 Will there be a separate PR to remove CPT_LINKS_PAGES?

Comment (nicoladj77) [reply app/modules/import-export/utils.php:84]: @louiswol94 yes I have it ready when this is merged.

File: core/document-types/page-base.php [hunk 3]
```diff
@@ -171,6 +171,29 @@ public static function register_style_controls( $document ) {
...
+			'view_item' => sprintf( __( 'View %s', 'elementor' ), $singular_label ),
+			'search_items' => sprintf( __( 'Search %s', 'elementor' ), $plural_label ),
+			'not_found' => sprintf( __( 'No %s found.', 'elementor' ), strtolower( $plural_label ) ),
+			'not_found_in_trash' => sprintf( __( 'No %s found in Trash.', 'elementor' ), strtolower( $plural_label ) ),
+			'parent_item_colon' => '',
+			'menu_name' => $plural_label,
+		];
+
+		return $labels;
+	}
```
Comment (louiswol94) [inline core/document-types/page-base.php:195]: @nicoladj77 Why did you have to add this here?

Comment (nicoladj77) [reply core/document-types/page-base.php:195]: @louiswol94 this is to streamline adding CPTs in modules. It's always the same so I thought it mades sense to have it here.

File: modules/conversion-center/assets/js/admin/links-pages.js [hunk 4]
```diff
@@ -27,16 +25,39 @@ export default class LinksPagesHandler extends AdminMenuHandler {
 		const settings = this.getSettings(),
 			isLinksPagesTablePage = !! window.location.href.includes( settings.paths.linksPagesTablePage ),
 			isLinksPagesTrashPage = !! window.location.href.includes( settings.paths.linksPagesTrashPage ),
-			isLinksPagesCreateYourFirstPage = !! window.location.href.includes( settings.paths.linksPagesAddNewPage );
+			isLinksPagesCreateYourFirstPage = !! window.location.href.includes( settings.paths.linksPagesAddNewPage ),
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply modules/conversion-center/assets/js/admin/links-pages.js:28]: @louiswol94 it will be renamed in the next PR where we will not have Link In bios doc anymore

File: modules/conversion-center/assets/js/editor/hooks/ui/editor/delete/open-library-after-delete.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ export class OpenLibraryAfterDelete extends After {
 			type = args?.containers[ 0 ]?.document?.config?.type;
 		}
 
-		return 'links-page' === type;
+		return 'links-page' === type || 'contact-buttons' === type;
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply modules/conversion-center/assets/js/editor/hooks/ui/editor/delete/open-library-after-delete.js:18]: @louiswol94 yes

File: modules/conversion-center/module.php [hunk 6]
```diff
@@ -296,35 +414,56 @@ private function register_links_pages_cpt() {
...
+			return $this->has_contact_pages;
+		}
+
+		$this->has_contact_pages = $this->has_pages(
+			self::CPT_CONTACT_PAGES,
+			self::CONTACT_PAGE_DOCUMENT_TYPE
+		);
+
+		return $this->has_contact_pages;
+	}
```
Comment (louiswol94) [inline modules/conversion-center/module.php:444]: We should probably split these into separate classes... 
`$this->register_contact_pages_cpt/$this->register_links_pages_cpt
`
--> 

`$this->links->register_cpt / $this->cb->register_cpt`

But we can do it if there's time left.

File: core/document-types/page-base.php (hunk 3 above)
Comment (louiswol94) [reply core/document-types/page-base.php:195]: It feels risky :)

File: modules/conversion-center/module.php (hunk 6 above)
Comment (nicoladj77) [reply modules/conversion-center/module.php:444]: @louiswol94 I think this should be architectured in a way that can be re used throughout the code base. Happy to give it a shot if we have time. Probably should live more in the Document rather than in the module, but not sure how/when modules are initialized.

**Review by louiswol94 — APPROVED** [review]
